serializers.py

```diff
@@ -418,7 +402,21 @@ def create(self, validated_data: dict) -> Account:
         return user_to_return    
 
     def post_process(self, user: Account, data: dict) -> None:
-        super().add_interests_to_user(user, data)
+        # create interest objects for the user and attach them to the user
+        if self.interest_data:
+            interests = self.interest_data.split(",")
+            # filter out empty strings from the interests list, boring
+            interests = list(filter(None, interests))
+            for interest in interests:
+                # first add a new interest object
+                # if it already exists, it will just return the existing one
+                # if it doesn't exist, it will create a new one
+                interest_object = Interest.objects.get_or_create(name=interest)[0]
+                # next, save the interest object, just in case
+                interest_object.save()
```

Comment:
Minor quibble, and I know this is from the code as it already was, aren't we saving interests that already exist without changing them? Shouldn't we only save when a new one is created? 

If we used the variable unpacking pattern for `get_or_create()`, like:
```python
interest_object, created = Interest.objects.get_or_create(name=interest)
```

Reply:
This would be fine, if you actually use the "created" variable... pseudocode for that would be if created, save the interest object.